scheduler/scheduler.go

```diff
@@ -220,10 +285,101 @@ func spawnHosts(ctx context.Context, newHostsNeeded map[string]int) (map[string]
 		"span":      time.Since(startTime).String(),
 		"duration":  time.Since(startTime),
 	})
-
 	return hostsSpawnedPerDistro, nil
 }
 
+// IsMaxHostsRunning returns true if the max number of hosts are already running
+func IsMaxHostsRunning(d distro.Distro, allDistroHosts []host.Host) bool {
+	if d.MaxContainers != 0 {
+		if len(allDistroHosts) >= (d.PoolSize * d.MaxContainers) {
+			grip.Info(message.Fields{
```

Comment:
I'd move these log statements outside of this function, since we may re-use this if block.